includes/admin/upgrades/upgrade-functions.php

```diff
@@ -1328,498 +1328,227 @@ function edd_include_file_download_log_migration_batch_processor( $class = '' )
-					<?php endif; ?>
-
-					<input type="hidden" name="edd-export-class" value="EDD\Admin\Upgrades\v3\Order_Notes" />
-				</span>
-				</form>
-			</div><!-- .inside -->
-		</div><!-- .postbox -->
+	<div id="edd-v3-migration-complete" <?php echo ! $can_remove_legacy_data ? 'class="edd-hidden"' : ''; ?>>
+		<p>
+			<?php esc_html_e( 'The data migration has been successfully completed. You may now leave this page or proceed to remove legacy data below.', 'easy-digital-downloads' ); ?>
+		</p>
 	</div>
-	<?php $step++; ?>
 
-	<div class="metabox-holder">
-		<div class="postbox">
-			<h2 class="hndle">
-				<span>
-					<?php echo esc_html( sprintf( /* translators: %d Migration step count */ __( 'Step %d: Upgrade Customer Notes', 'easy-digital-downloads' ), $step ) ); ?>
-				</span>
-				<span class="dashicons dashicons-yes"></span>
-			</h2>
-			<div class="inside migrate-customer-notes-control">
-				<p>
-					<?php esc_html_e( 'This will migrate all customer notes from the WordPress posts table to custom database tables for improved performance and reliability.', 'easy-digital-downloads' ); ?>
-				</p>
-				<form method="post" id="edd-migrate-customer-notes-form" class="edd-export-form edd-import-export-form">
-				<span class="step-instructions-wrapper">
-					<?php wp_nonce_field( 'edd_ajax_export', 'edd_ajax_export' ); ?>
-
-					<?php if ( ! $customer_notes_removal_complete ) : ?>
-						<span class="edd-migration allowed" style="<?php echo $migration_complete ? '' : 'display: none'; ?>">
-							<button type="submit" id="migrate-customer-notes-submit" class="button button-primary"><?php esc_attr_e( 'Upgrade Database', 'easy-digital-downloads' ); ?></button>
-						</span>
-
-						<span class="edd-migration unavailable" style="<?php echo ! $migration_complete ? '' : 'display: none'; ?>">
-							<button type="submit" disabled id="migrate-customer-notes-submit" class="button button-secondary"><?php esc_attr_e( 'Upgrade Database', 'easy-digital-downloads' ); ?></button>
-							&mdash;
-								 <span class="message initial"><?php esc_html_e( 'Please complete the previous steps before updating customer notes.', 'easy-digital-downloads' ); ?></span>
-								 <span class="message complete"><?php esc_html_e( 'Data migration complete.', 'easy-digital-downloads' ); ?></span>
-						</span>
-					<?php else: ?>
-						<button type="submit" disabled id="migrate-customer-notes-submit" class="button button-secondary"><?php esc_attr_e( 'Upgrade Database', 'easy-digital-downloads' ); ?></button>
-						&mdash; <?php esc_html_e( 'Legacy data has already been removed, migration is not possible at this time.', 'easy-digital-downloads' ); ?>
-					<?php endif; ?>
-
-					<input type="hidden" name="edd-export-class" value="EDD\Admin\Upgrades\v3\Customer_Notes" />
+	<div id="edd-v3-remove-legacy-data" <?php echo ! $can_remove_legacy_data ? 'class="edd-hidden"' : ''; ?>>
+		<h2><?php esc_html_e( 'Remove Legacy Data', 'easy-digital-downloads' ); ?></h2>
```

Comment:
After I've removed the legacy data, I can reload the page and still get this section telling me to remove it. Could we add a check and say something like "Legacy data has already been removed; you're good to go!"?

Reply:
Yeah still working on fine tuning that logic.